app/src/configure-qvm.lisp

```diff
@@ -63,33 +65,37 @@
          (error "Cannot simultaneously support Pauli error model and Kraus channels."))
 
        (multiple-value-bind (amps fin) (make-wavefunction (expt 2 num-qubits))
-         (let ((q
-                 (make-instance 'qvm:noisy-qvm
-                                :number-of-qubits num-qubits
-                                :noisy-gate-definitions kraus-ops
-                                :readout-povms readout-povms
-                                :classical-memory-subsystem (make-instance 'qvm:classical-memory-subsystem
-                                                                           :classical-memory-model
-                                                                           classical-memory-model)
-                                :amplitudes amps)))
+         (let* ((state (make-instance 'qvm:pure-state :num-qubits num-qubits
+                                                      :amplitudes amps))
+                (q
+                  (make-instance 'qvm:noisy-qvm
+                                 :number-of-qubits num-qubits
+                                 :noisy-gate-definitions kraus-ops
+                                 :readout-povms readout-povms
+                                 :classical-memory-subsystem (make-instance 'qvm:classical-memory-subsystem
+                                                                            :classical-memory-model
+                                                                            classical-memory-model)
+                                 :state state)))
            (when fin (tg:finalize q fin))
```

Comment:
would it make more sense to `(tg:finalize state fin)` here since the `state` object now "owns" the amplitudes?